### DISCUSSION TITLE
The update doesn't seem to be working very well at the moment

### DISCUSSION THREAD

**Discussion by Ran-Xing** [coder/code-server#4316] [Q&A]
```ini
[root@c63fc09e96eb ~]# cd /root && yum install wget -y  && wget https://github.com/cdr/code-server/releases/download/v3.9.0/code-server-3.9.0-amd64.rpm && rpm -ivh code-server-3.9.0-amd64.rpm
Nothing to do.
Complete!
--2021-10-06 21:21:22--  https://github.com/cdr/code-server/releases/download/v3.9.0/code-server-3.9.0-amd64.rpm
Resolving github.com (github.com)...
Saving to: 'code-server-3.9.0-amd64.rpm'
Verifying...                          ################################# [100%]
Preparing...                          ################################# [100%]

[root@c63fc09e96eb ~]# /usr/bin/code-server -v
[2021-10-06T21:22:13.366Z] info  Wrote default config file to ~/.config/code-server/config.yaml
3.9.0 fc6d123da59a4e5a675ac8e080f66e032ba01a1b

[root@c63fc09e96eb ~]# wget https://github.com/cdr/code-server/releases/download/v3.12.0/code-server-3.12.0-amd64.rpm
--2021-10-06 21:22:34--  https://github.com/cdr/code-server/releases/download/v3.12.0/code-server-3.12.0-amd64.rpm
Resolving github.com (github.com)...
Saving to: 'code-server-3.12.0-amd64.rpm'
Verifying...                          ################################# [100%]
Preparing...                          ################################# [100%]
	file /usr/lib/systemd/user/code-server.service from install of code-server-0:3.12.0-1.x86_64 conflicts with file from package code-server-0:3.9.0-1.x86_64

[root@c63fc09e96eb ~]# /usr/bin/code-server -v
3.9.0 fc6d123da59a4e5a675ac8e080f66e032ba01a1b
```

**Comment by jsjoeio** [comment]
> 	file /usr/lib/systemd/user/code-server.service from install of code-server-0:3.12.0-1.x86_64 conflicts with file from package code-server-0:3.9.0-1.x86_64

I'm not super familiar with our code-server service. @code-asher is this a bug?

Otherwise, do you need to [remove the service](https://amalgjose.com/2020/07/17/how-to-remove-a-systemd-service-from-linux-centos-rhel-ubuntu/) and then upgrade code-server?

**Reply by code-asher** [reply]
I am not sure if we have an issue for it yet but it is probably the same as https://github.com/cdr/code-server/discussions/4191

**Reply by code-asher** [reply]
Oh just saw this does not use our install script.  In that case use `-U` instead of `-i`.  But we do need to also fix our install script.

**Reply by jsjoeio** [reply]
Filed an issue: https://github.com/cdr/code-server/issues/4323

**Reply by lg-nineoneninethreenine** [reply]
Has anyone tried deleting the 3.9.0 RPM download and then downloaded 3.12.0 RPM?

**Reply by Ran-Xing** [reply]
@code-asher Are you sure that only the installation script cannot be updated?
It seems that rpm -ivh cannot be updated either.
Are you talking about rpm -uvh?
My English is not very good, please forgive me.

**Reply by code-asher** [reply]
Yes that is right! `rpm -i` will not work to update an rpm package, you will have to use `rpm -U`

http://ftp.rpm.org/max-rpm/ch-rpm-upgrade.html

As @im-coder-lg mentioned you can also remove the current code-server (`-e` I think?) and then use `-i` to install the new one.
